src/libs/ReportUtils.js

```diff
@@ -1324,12 +1324,27 @@ function getLastVisibleMessage(reportID, actionsToMerge = {}) {
 }
 
 /**
- * Determines if a report has an IOU that is waiting for an action from the current user (either Pay or Add a credit bank account)
+ * Checks if a report is an open task report assigned to current user.
+ *
+ * @param {Object} report
+ * @param {Object} parentReportAction - The parent report action of the report (Used to check if the task has been canceled)
+ * @returns {Boolean}
+ */
+function isWaitingForTaskCompleteFromAssignee(report, parentReportAction = {}) {
+    return isTaskReport(report) && isReportManager(report) && isOpenTaskReport(report, parentReportAction);
+}
+
+/**
+ * Determines if a report should show a GBR (green dot) in the LHN. This can happen when the report: 
+    - is unread and the user was mentioned in one of the unread comments
+    - is for an outstanding task waiting on the user
+    - has an outstanding child money request that is waiting for an action from the current user (e.g. pay, approve, add bank account)
  *
  * @param {Object} report (chatReport or iouReport)
+ * @param {Object} parentReportAction (the report action the current report is a thread of)
  * @returns {boolean}
  */
-function isWaitingForIOUActionFromCurrentUser(report) {
+function shouldShowGBR(report, parentReportAction = {}) {
```

Comment:
NAB, I think we could separate the UI concerns from this logic a bit. A component (or non-UI code) might decide to do things with the same info. so, could maybe do something like `ReportUtils.requiresAttentionFromCurrentUser()`.

Reply:
Ok I like that. It is used in at least one place right now that isn't directly about deciding whether to show a GBR.